bombastic/api/src/server.rs

```diff
@@ -278,13 +279,13 @@ async fn publish_sbom(
     let typ = verify_type(content_type)?;
     let enc = verify_encoding(req.headers().get(CONTENT_ENCODING))?;
     let id = &params.id;
-    let mut payload = payload.map_err(|e| match e {
-        PayloadError::Io(e) => e,
-        _ => io::Error::new(io::ErrorKind::Other, e),
+    let payload = payload.map_err(|e| match e {
+        PayloadError::Io(e) => StorageError::Io(e),
+        _ => StorageError::Io(io::Error::new(io::ErrorKind::Other, e)),
     });
     let size = state
         .storage
-        .put_stream(id, typ.as_ref(), enc, &mut payload)
+        .put_stream(id, typ.as_ref(), enc, payload)
```

Comment:
Here... payload may still be `Err` variant, yeah?

Reply:
No, it's a virgin stream at this point. `put_stream` will attempt to read from it and if it gets an error, that fn above will be invoked.